File: w02/d02/gadi_gottlieb/class_work/pets/lib/pet.rb
require 'pry'

class Pet

  attr_reader(:name, :age, :owner, :foods_eaten)

  def initialize(name, age, owner)
    @name = name
    @age = age
    @owner = owner
    @foods_eaten = []
  end

  def love_owner()
    return "I love my owner #{owner}"
  end

  def eat(food)
    @foods_eaten.push(food)
  end

  def list_foods()
    return @foods_eaten.join(', ')
  end

end






# * Pet
# * name - string
# * owner - string, owner's name
# * age - number
# * foods_eaten - initialized to empty array
# * love_o)wner() - return a string proclaiming love towards owner
# * eat(food) - push 'food' into the foods_eaten array
# * list_foods() - return a string containing all the eaten foods
# * Dog
# * ear_type - A string (Eg. "Floopy", "Upright", "Stubby", "Shaggy")
# * bark() - Returns a string with a bark in it (eg. "Woof woof")
# * Cat
# * hours_slept - initialized to 0
# * sleep() - increments hours_slept by 1
# * judge_person(person) - Accepts a string 'person' and returns a judgmental string about that person

File: w10/d05/najee_gardner/inheretance/pet.js
function Pet(options) {
  this.name = options.name;
  this.owner = options.owner;
}
Pet.prototype.introduce = function() {
  return 'Hello, I am ' + this.name;
}

function Cat(options) {
  Pet.call(this, options);
  this.favFood = options.favFood;
}
Cat.prototype = Object.create(Pet.prototype);
Cat.prototype.judgePerson = function() {
  return 'You aren\'t so great, ' + this.owner;
};

function Dog(options) {
  // this.constructor(name, owner);
  Pet.call(this, options);
  this.earType = options.earType;
}
Dog.prototype = Object.create(Pet.prototype);
Dog.prototype.bark = function() {
  return 'Woof!';
};
